File: frontend/alina-frontend/src/lib/api/types/revision.ts
// Revision Request types

export enum RevisionStatus {
  REQUESTED = 'Requested',
  IN_PROGRESS = 'InProgress',
  SUBMITTED = 'Submitted',
  ACCEPTED = 'Accepted',
  REJECTED = 'Rejected',
}

export interface Revision {
  id: string;
  orderId: string;
  requestedBy: string;
  requestedByName?: string;
  description: string;
  status: RevisionStatus;
  attachments?: string[];
  submittedAt?: string;
  acceptedAt?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateRevisionRequest {
  orderId: string;
  description: string;
  attachments?: File[];
}

export interface SubmitRevisionRequest {
  revisionId: string;
  description?: string;
  attachments?: File[];
}

File: backend/alina-backend/Modules/orders/Revision.cs
using System.ComponentModel.DataAnnotations;
using alina_backend.Modules.users;

namespace alina_backend.Modules.orders;

public enum RevisionStatus
{
    Requested,
    Accepted,
    Rejected,
    Completed
}

public class Revision
{
    public int Id { get; set; }

    [Required]
    public int OrderId { get; set; }

    [Required]
    public Guid RequesterId { get; set; }
    public User? Requester { get; set; }

    [Required]
    public string Description { get; set; } = string.Empty;

    public string? Attachments { get; set; } // JSON array of attachment URLs

    public RevisionStatus Status { get; set; } = RevisionStatus.Requested;

    public string? ResolutionMessage { get; set; }

    public string? ResolutionAttachments { get; set; } // JSON array of resolution attachment URLs

    public DateTime RequestedAt { get; set; } = DateTime.UtcNow;

    public DateTime? RespondedAt { get; set; }
}